### ISSUE TITLE
Version mismatch

### ISSUE THREAD

**Issue by elbaro** [Homebrew/homebrew-core#39521]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [X] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
- [X] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [X] ran `brew update` and can still reproduce the problem?
- [X] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [X] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [X] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

```sh
/usr/local/Cellar
❯ brew info libpng
libpng: stable 1.6.37 (bottled), HEAD
Library for manipulating PNG images
http://www.libpng.org/pub/png/libpng.html
/usr/local/Cellar/libpng/1.6.37 (27 files, 1.2MB) *
  Poured from bottle on 2019-04-20 at 18:06:16
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/libpng.rb
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 255,346 (30 days), 604,005 (90 days), 2,083,880 (365 days)
install_on_request: 14,145 (30 days), 30,685 (90 days), 105,461 (365 days)
build_error: 0 (30 days)

/usr/local/Cellar
❯ rg 1.6.36/
freetype/2.10.0/bin/freetype-config
48:  static_libs="-lfreetype -lz -lbz2 -L/usr/local/Cellar/libpng/1.6.36/lib -lpng16 -lz"

freetype/2.10.0/lib/pkgconfig/freetype2.pc
13:Libs.private: -lz -lbz2 -L/usr/local/Cellar/libpng/1.6.36/lib -lpng16 -lz

leptonica/1.78.0/lib/pkgconfig/lept.pc
10:Libs.private: -lz -L/usr/local/Cellar/libpng/1.6.36/lib -lpng16 -lz -L/usr/local/Cellar/jpeg/9c/lib -ljpeg -lgif -L/usr/local/Cellar/libtiff/4.0.10_1/lib -ltiff -L/usr/local/Cellar/webp/1.0.2/lib -lwebp

```

- libpng version is 1.6.37
- some packages refer to libpng 1.6.36
- brew is up-to-date

**Comment by SMillerDev** [comment]
Yeah, all those packages pointing to 1.6.36 need to be recompiled on your machine. Your best option is not to install things with the HEAD option unless you know what you are doing and really need the very latest features before release.

**Comment by elbaro** [comment]
@SMillerDev I didn't provide `--HEAD` option. libpng and freetype are installed with no extra arguments.

Uninstall
```
❯ brew uninstall --ignore-dependencies libpng
Uninstalling /usr/local/Cellar/libpng/1.6.37... (27 files, 1.2MB)
```

Reinstall
```
❯ brew install libpng
==> Downloading https://homebrew.bintray.com/bottles/libpng-1.6.37.mojave.bottle.tar.gz
Already downloaded: /Users/emppu/Library/Caches/Homebrew/downloads/1957fe4350dafd271e0a5ae9ac4c9e3583520b3b77111f873366c8a21f2e9703--libpng-1.6.37.mojave.bottle.tar.gz
==> Pouring libpng-1.6.37.mojave.bottle.tar.gz
🍺  /usr/local/Cellar/libpng/1.6.37: 27 files, 1.2MB
```

Still --HEAD
```
❯ brew info libpng
libpng: stable 1.6.37 (bottled), HEAD
Library for manipulating PNG images
http://www.libpng.org/pub/png/libpng.html
/usr/local/Cellar/libpng/1.6.37 (27 files, 1.2MB) *
  Poured from bottle on 2019-05-05 at 20:35:36
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/libpng.rb
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 253,742 (30 days), 604,292 (90 days), 2,084,486 (365 days)
install_on_request: 14,190 (30 days), 30,704 (90 days), 105,568 (365 days)
build_error: 0 (30 days)

```

**Comment by fxcoudert** [comment]
I've launched a rebuild of the bottles at https://jenkins.brew.sh/job/Homebrew%20Testing/1173/
This will fix the issue.

**Comment by fxcoudert** [comment]
Many tests failed when rebottling, which is weird (and I haven't had time to investigate). A revision bump PR is probably the next thing to try.

**Comment by zbeekman** [comment]
This still seems to be a problem for me. I'll try rebuilding locally to see if I can figure out what's up.

**Comment by fxcoudert** [comment]
Fixed in ec4355417c6358058f733ab95cdf554878d59a07
